Assets/Tests/Runtime/NetworkedMessageChannelTests.cs

```diff
@@ -0,0 +1,204 @@
+    {
+        struct EmptyMessage : INetworkSerializeByMemcpy { }
+
+        struct GenericMessage : INetworkSerializeByMemcpy
+        {
+            public bool value;
+        }
+
+        protected override int NumberOfClients => 2;
+
+        NetworkManager FirstClient => m_ClientNetworkManagers[0];
+        NetworkManager SecondClient => m_ClientNetworkManagers[1];
+
+        DisposableGroup m_Subscriptions;
+
+        DIScope m_FirstClientScope;
+        DIScope m_SecondClientScope;
+        DIScope m_ServerScope;
+
+        protected override void OnServerAndClientsCreated()
+        {
+            m_FirstClientScope = new DIScope();
+            m_FirstClientScope.BindInstanceAsSingle(FirstClient);
+            m_FirstClientScope.FinalizeScopeConstruction();
+
+            m_SecondClientScope = new DIScope();
+            m_SecondClientScope.BindInstanceAsSingle(SecondClient);
+            m_SecondClientScope.FinalizeScopeConstruction();
+
+            m_ServerScope = new DIScope();
+            m_ServerScope.BindInstanceAsSingle(m_ServerNetworkManager);
+            m_ServerScope.FinalizeScopeConstruction();
+
+            base.OnServerAndClientsCreated();
+        }
+
+        protected override IEnumerator OnTearDown()
+        {
+            m_Subscriptions.Dispose();
+            m_FirstClientScope.Dispose();
+            m_SecondClientScope.Dispose();
+            m_ServerScope.Dispose();
+            return base.OnTearDown();
+        }
+
+
+        [UnityTest]
+        public IEnumerator EmptyNetworkedMessageIsReceivedByAllSubscribersOnMultipleClients()
+        {
+            var emptyMessageChannelClient1 = new NetworkedMessageChannel<EmptyMessage>();
+            m_FirstClientScope.InjectIn(emptyMessageChannelClient1);
+            var emptyMessageChannelClient2 = new NetworkedMessageChannel<EmptyMessage>();
+            m_SecondClientScope.InjectIn(emptyMessageChannelClient2);
+            var emptyMessageChannelServer = new NetworkedMessageChannel<EmptyMessage>();
+            m_ServerScope.InjectIn(emptyMessageChannelServer);
+
+            var nbMessagesReceived = 0;
+
+            m_Subscriptions = new DisposableGroup();
+            m_Subscriptions.Add(emptyMessageChannelClient1.Subscribe(OnEmptyMessageReceived));
+            m_Subscriptions.Add(emptyMessageChannelClient2.Subscribe(OnEmptyMessageReceived));
+
+            void OnEmptyMessageReceived(EmptyMessage message)
+            {
+                nbMessagesReceived++;
+            }
+
+            emptyMessageChannelServer.Publish(new EmptyMessage());
+
+            // wait for the custom named message to be sent on the server and received on the clients
+            yield return null;
+            yield return null;
+
+            Assert.AreEqual(2, nbMessagesReceived);
+        }
+
+        [UnityTest]
+        public IEnumerator EmptyNetworkedMessageIsReceivedByAllSubscribersOnSingleClient()
```

Comment:
Since this seems like a copy of the above test, you can parameterize tests and specify "number of clients to spawn" for example and use only one test

Reply:
It's not exactly the same though. Above, we have two clients with one subscription each, and below we have one client with two subscriptions.

Reply:
but if you think it'll add too much complexity, you can disregard this comment and resolve it.

Reply:
I should have replied to that comment after updating the PR. In the end, I simply parameterised both the number of clients and the number of subscriptions per client, as well as the message type. So yes this can be resolved